x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/MultivalueTestCaseSupplier.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function;
+
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import static org.elasticsearch.test.ESTestCase.randomBoolean;
+import static org.elasticsearch.test.ESTestCase.randomList;
+import static org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier.TypedDataSupplier;
+
+/**
+ * Extension of {@link TestCaseSupplier} that provided multivalue test cases.
+ */
+public final class MultivalueTestCaseSupplier {
+
+    private static final int MIN_VALUES = 1;
+    private static final int MAX_VALUES = 1000;
+
+    private MultivalueTestCaseSupplier() {}
+
+    public static List<TypedDataSupplier> intCases(int min, int max, boolean includeZero) {
+        List<TypedDataSupplier> cases = new ArrayList<>();
+
+        for (Block.MvOrdering ordering : Block.MvOrdering.values()) {
```

Comment:
I wonder if those are too much...

Reply:
Interesting! I can see why you'd do it. What's the runtime difference on one of the tests? I can imagine it's a few seconds so it's worth it.

Another option is to randomize this bit. That's less cases at least.

Reply:
On second inspection `AbstractMultivalueTestCase`'s generators already have this. So I'd just keep it and not worry about it. For MV functions it's quite an important bit of their behavior.